proofs/signatures/drat.plf

```diff
@@ -29,10 +29,20 @@
 ; We could sort them, but that would require defining an order on our variables,
 ; and incurring the cost of sorting.
 ; Instead, we do the following:
-;  1. Sweep the first clause, marking variables with flags 1 (pos) and 2 (neg)
-;  2. Sweep the second clause, erasing marks.
+;  1. Sweep the first clause, marking variables with flags 1,3 (pos) and 2,4 (neg)
+;  2. Sweep the second clause, erasing marks 1,2.
 ;  3. Unsweep the first clause, returning FALSE on marks.
-;     Also unmarking
+;     Also unmarking 3,4, and 1,2 if needed
+;
+; So the mark 1 means (seen as + in c1, but not yet in c2)
+;    the mark 3 means (seen as + in c1)
+;    the mark 2 means (seen as - in c1, but not yet in c2)
+;    the mark 4 means (seen as - in c1)
+;
+; This implementation is robust to literal order, repeated literals, and
+; literals of opposite polarity in the same clause. It is true equality on sets
+; literals.
+;
 ; TODO(aozdemir) This implementation could be further optimized b/c once c1 is
 ; drained, we need not continue to pattern match on it.
 (program clause_eq ((c1 clause) (c2 clause)) bool
```

Comment:
Does this program assumes marks 1,2,3,4 are unset upon entry, and unsets them upon exit? If so, maybe worth documenting.

Reply:
The program does make those assumptions, and they should be documented.

Thanks!